tsdb/shard.go

```diff
@@ -40,6 +40,7 @@ type Shard struct {
 	db    *bolt.DB // underlying data store
 	index *DatabaseIndex
 	path  string
+	id    uint64
```

Comment:
because we need to track which series in the index exist in which shards so we know when metadata needs to be persisted to them. See https://github.com/influxdb/influxdb/blob/master/tsdb/shard.go#L331-L338

This was the fix for that rather nasty bug I mentioned in standup on Monday.